// @sap-cloud-sdk/odata-v2/dist/request-builder/response-transformers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.transformReturnValueForEdmTypeList = exports.transformReturnValueForEdmType = exports.transformReturnValueForComplexTypeList = exports.transformReturnValueForComplexType = exports.transformReturnValueForEntityList = exports.transformReturnValueForEntity = exports.transformReturnValueForUndefined = void 0;
const de_serializers_1 = require("../de-serializers");
const response_data_accessor_1 = require("./response-data-accessor");
function transformReturnValueForUndefined(data, builderFn) {
    return builderFn(data);
}
exports.transformReturnValueForUndefined = transformReturnValueForUndefined;
function transformReturnValueForEntity(data, entityApi) {
    return (0, de_serializers_1.entityDeserializer)(entityApi.deSerializers)
        .deserializeEntity((0, response_data_accessor_1.getSingleResult)(data), entityApi)
        .setOrInitializeRemoteState();
}
exports.transformReturnValueForEntity = transformReturnValueForEntity;
function transformReturnValueForEntityList(data, entityApi) {
    const deserializeEntity = (0, de_serializers_1.entityDeserializer)(entityApi.deSerializers).deserializeEntity;
    return (0, response_data_accessor_1.getCollectionResult)(data).map(entityJson => deserializeEntity(entityJson, entityApi).setOrInitializeRemoteState());
}
exports.transformReturnValueForEntityList = transformReturnValueForEntityList;
function transformReturnValueForComplexType(data, builderFn) {
    return builderFn((0, response_data_accessor_1.getSingleResult)(data));
}
exports.transformReturnValueForComplexType = transformReturnValueForComplexType;
function transformReturnValueForComplexTypeList(data, builderFn) {
    return (0, response_data_accessor_1.getCollectionResult)(data).map(json => builderFn(json));
}
exports.transformReturnValueForComplexTypeList = transformReturnValueForComplexTypeList;
function transformReturnValueForEdmType(data, builderFn) {
    return builderFn((0, response_data_accessor_1.getSingleResult)(data));
}
exports.transformReturnValueForEdmType = transformReturnValueForEdmType;
function transformReturnValueForEdmTypeList(data, builderFn) {
    return (0, response_data_accessor_1.getCollectionResult)(data).map(builderFn);
}
exports.transformReturnValueForEdmTypeList = transformReturnValueForEdmTypeList;
